tslearn/shapelets.py

```diff
@@ -410,12 +444,13 @@ def fit(self, X, y):
 
         self._set_model_layers(X=X, ts_sz=sz, d=d, n_classes=n_labels)
         self._set_weights_false_conv(d=d)
-        self.model_.fit(
+        h = self.model_.fit(
             [X[:, :, di].reshape((n_ts, sz, 1)) for di in range(d)], y_,
             batch_size=self.batch_size, epochs=self.max_iter,
             verbose=self.verbose
         )
-        self.n_iter_ = len(self.model_.history.history)
+        self.history_ = h.history
```

Comment:
Are there any other interesting attributes in `h`? Perhaps we should store this instead of only the history?

Reply:
I've never used anything else personally. And `keras` docs (description of the `fit` method) seem to only stress out that dict:

```
Returns

A History object. Its History.history attribute is a record of training loss values and metrics values at successive epochs, as well as validation loss values and validation metrics values (if applicable).
```

Finally, a dict is much easier to serialize than a custom class, which is why I made this choice.